### PR TITLE
Simplification of the contracts 

### PR DESCRIPTION
**Pull request by nivida**

### CODE REVIEW

File: packages/contracts/contracts/core/processes/Process.sol [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+/// @title Abstract implementation of the governance process
+/// @author Samuel Furter - Aragon Association - 2021
+/// @notice This contract can be used to implement concrete governance processes and being fully compatible with the DAO framework and UI of Aragon
+/// @dev You only have to define the specific custom logic for your needs in _start and _execute.
+abstract contract Process is Component {
+    address internal constant ANY_ADDR = address(type(uint160).max);
+
+    // Events
+    event ProcessStarted(Execution indexed execution, uint256 indexed executionId, bytes indexed metadata);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/processes/Process.sol]: can we remove indexed? for Execution

File: packages/contracts/contracts/core/processes/Process.sol [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    event ProcessExecuted(Execution indexed execution, uint256 indexed executionId);
+    event AllowedActionsAdded(bytes[] indexed allowedActions);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/processes/Process.sol]: remove indexed for array, and Execution

File: packages/contracts/contracts/core/processes/disputable/DisputableProcess.sol [hunk 3]
```diff
@@ -2,56 +2,52 @@
...
 /// @author Samuel Furter - Aragon Association - 2021
-/// @notice This contract can be used to implement concrete disputable governance primitives and being fully compatible with the DAO framework and UI of Aragon
+/// @notice This contract can be used to implement concrete disputable governance processes and being fully compatible with the DAO framework and UI of Aragon
 /// @dev You only have to define the specific custom logic of your needs in _start, _execute, _halt, and _forward
-abstract contract DisputableGovernancePrimitive is StoppableGovernancePrimitive {
-    event GovernancePrimitiveHalted(Execution indexed execution, uint256 indexed executionId);
-    event GovernancePrimitiveForwarded(Execution indexed execution, uint256 indexed executionId);
+abstract contract DisputableProcess is StoppableProcess {
+    event ProcessHalted(Execution indexed execution, uint256 indexed executionId);
+    event ProcessForwarded(Execution indexed execution, uint256 indexed executionId);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/processes/disputable/DisputableProcess.sol]: remove indexed

File: packages/contracts/contracts/core/processes/stoppable/StoppableProcess.sol [hunk 4]
```diff
@@ -2,35 +2,34 @@
...
-/// @title Abstract implementation of the stoppable governance primitive
+/// @title Abstract implementation of the stoppable governance process
 /// @author Samuel Furter - Aragon Association - 2021
-/// @notice This contract can be used to implement concrete stoppable governance primitives and being fully compatible with the DAO framework and UI of Aragon
+/// @notice This contract can be used to implement concrete stoppable governance processes and being fully compatible with the DAO framework and UI of Aragon
 /// @dev You only have to define the specific custom logic for your needs in _start, _execute, and _stop
-abstract contract StoppableGovernancePrimitive is GovernancePrimitive {
-    event GovernancePrimitiveStopped(Execution indexed execution, uint256 indexed executionId);
+abstract contract StoppableProcess is Process {
+    event ProcessStopped(Execution indexed execution, uint256 indexed executionId);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/processes/stoppable/StoppableProcess.sol:14]: remove indexed

File: packages/contracts/contracts/core/processes/voting/VotingProcess.sol [hunk 5]
```diff
@@ -2,32 +2,31 @@
...
 
-/// @title Abstract implementation of the voting governance primitive
+/// @title Abstract implementation of the voting governance process
 /// @author Samuel Furter - Aragon Association - 2021
 /// @notice This contract can be used to implement concrete voting governance primitives and being fully compatible with the DAO framework and UI of Aragon
 /// @dev You only have to define the specific custom logic of your needs in _vote, _start, and _execute
-abstract contract VotingGovernancePrimitive is GovernancePrimitive {
-    event VotedOnGovernancePrimitive(Execution indexed execution, bytes indexed data, uint256 executionId);
+abstract contract VotingProcess is Process {
+    event VotedOnProcess(Execution indexed execution, bytes indexed data, uint256 executionId);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/processes/voting/VotingProcess.sol]: remove indexed

File: packages/contracts/contracts/registry/Registry.sol [hunk 6]
```diff
@@ -2,27 +2,27 @@
...
 
-import "../DAO.sol";
+import "../core/DAO.sol";
 
 /// @title Register your unique DAO name
 /// @author Samuel Furter - Aragon Association - 2021
 /// @notice This contract provides the possiblity to register a DAO by a unique name.
 contract Registry {
-    event NewDAO(string indexed name, DAO indexed dao, address indexed creator);
+    event NewDAORegistered(string indexed name, DAO indexed dao, address indexed creator);
```
Comment (Rekard0) [inline packages/contracts/contracts/registry/Registry.sol]: remove indexed for string

File: packages/contracts/contracts/core/DAO.sol [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    /// @param _metadata The IPFS hash of the new metadata object
+    function setMetadata(bytes calldata _metadata) external auth(address(this), DAO_CONFIG_ROLE) {
+        emit SetMetadata(_metadata);
+    }
+
+    /// @notice Add new process to DAO
+    /// @dev Grants the new process execution rights and amits the related event.
+    /// @param process The address of the new process
+    function addProcess(Process process) external auth(address(this), DAO_CONFIG_ROLE) {
+        _grant(address(this), address(process), EXEC_ROLE);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/DAO.sol:90]: should we have some sort of check (for selectors perhaps) to check if process follows Aragon's core process or it is alien?

Comment (nivida) [reply packages/contracts/contracts/core/DAO.sol:90]: Yeah, have written it down to implement EIP-165. Will most likely also be required on your side in the subgraph.

**Comment by nivida** [comment]
I've merged this already cause the PR is anyways way to big. And it was a complete rewrite of the PoC contracts we had. The refactoring happened to improve the readability of the framework and to significantly lower the complexity of the core.

**Pull request merged**